src/useSelect.js

```diff
@@ -77,7 +77,9 @@ export default function useSelectSearch({
     const onFocus = () => setFocus(true);
     const onSelect = useCallback((val) => {
         setState((oldState) => {
-            const item = val || oldState.flat[oldState.highlighted].value;
+            const defaultItem = oldState.flat[oldState.highlighted];
+            const oldStateValue = defaultItem && defaultItem.value;
+            const item = val || oldStateValue;
```

Comment:
The `item` here will still be undefined and therefor crash when it's sent to getOption (because `values` will be an array of undefined values).

I think we need to:

* Exit if item is undefined
* Enter should not trigger onSelect if highlighted is -1
* Filter the values array in getOption before we map it